### PR TITLE
Astro (SSR + DB) + Htmx + Netlify Blogpost

### PR DESCRIPTION
**Pull request by flyck**
Remaining tasks:
- [x] find fitting post slug url
- [x] resolve linting errors (see related MR https://github.com/hendriknielaender/double-trouble/pull/454)

### CODE REVIEW

**Comment by hendriknielaender** [comment]
htmx1
htmx5
htmx4
htmx3
htmx2

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+---
+publishDate: "Apr 2 2024"
+title: "How to ditch React and build an app with HTMX, Astro DB & Astro SSR"
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:3]:
```suggestion
title: "Ditch React: Build Faster with htmx, Astro DB & Astro SSR in 2024"
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+A responsive button would change its appearance while the operation is ongoing. For our usecase
+this means we want to show a "Confirm" text while the button is ready, and a loading spinner while
+the confirmation is being processed.
+
+<img
+    style="display: block;
+           margin-left: auto;
+           margin-right: auto;"
+    src="../../src/assets/images/posts/howto-htmx-astrodb-astrossr/ezgif-button.gif">
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:35]:
```suggestion
  alt="Demonstration of HTMX responsive button"
src="../../src/assets/images/posts/howto-htmx-astrodb-astrossr/ezgif-button.gif">
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+title: "How to ditch React and build an app with HTMX, Astro DB & Astro SSR"
+description: "Explore common techniques and solutions needed for building a basic CRUD app with this modern react-less stack."
+image: "~/assets/images/thumbnails/htmx_app.jpg"
+imageCreditUrl: https://midjourney.com
+tags: [htmx, react, astro, astrodb, astrossr, crud, netlify]
+---
+
+These days, many people feel tired of React, and as [Primeagen](https://twitter.com/ThePrimeagen)
+announced it: `This is the year of HTMX!` While the htmx examples in the documentation look easy
+enough, we took it to the test and and re-implement an existing React project in HTMX.
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:12]:
```suggestion
enough, we took it to the test and re-implemented an existing React project in htmx.
```

File: data/blog/howto-htmx-astrodb-astrossr.md (hunk 1 above)
Comment (hendriknielaender) [reply data/blog/howto-htmx-astrodb-astrossr.md:3]: Ensure htmx is always lowercase. https://htmx.org/

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+We had created the original app earlier this year mostly to play around with Websockets and
+AppSync subcriptions. In the frontend it was based on React and Nextjs, hosted ontop of Vercel. On
+the backend it ran on AWS AppSync and Dynamodb as a persistent storage. In the re-implementation
+we moved the frontend from to purely html, using [HTMX](https://htmx.org/) and
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:17]:
```suggestion
we moved the frontend from to purely html, using [htmx](https://htmx.org/ "Visit HTMX Official Site for more details") and
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+
+We had created the original app earlier this year, mostly to play around with Websockets and
+AppSync subscriptions. In the frontend it was based on React and Next.js, hosted on top of
+Vercel. On the backend it ran on AWS AppSync and Dynamodb as a persistent storage. In the
+[re-implementation](https://github.com/flyck/astro-party), we moved the frontend to pure HTML and
+js, using [htmx](https://htmx.org/ "Visit HTMX Official Site for more details") and
+[Astro](https://astro.build). For the backend we used [Astro DB](https://astro.build/db/), which
+is currently in early preview, and Astro SSR hosted on [Netlify](https://www.netlify.com/).
+
+We were curious how well this stack could could work for CRUD apps, but especially the experience
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:29]:
```suggestion
We were curious how well this stack could work for CRUD apps, but especially the experience
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+
+<img
+    style="display: block;
+           margin-left: auto;
+           margin-right: auto;"
+  alt="Demonstration of HTMX responsive button"
+src="../../src/assets/images/posts/howto-htmx-astrodb-astrossr/button.gif">
+</img>
+
+Htmx automatically adds predefined CSS classes to requesting elements, so with tailwinds
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:47]:
```suggestion 
Htmx automatically adds predefined CSS classes to requesting elements, so with tailwind's
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+the same while only fetching data once, had us come up with some last-mile JavaScript. The
+alternative could be to simply fetch more HTML, which seems to be part of the htmx paradigm.
+
+Generally the chance of writing explicit JavaScript seems slightly higher with htmx, but at-least
+you know what you're doing, and it's very honest work. Though in this case more explicit JavaScript
+was written, the perceived complexity was lower and the overall enjoyment greater.
+
+## Rounding up
+
+Coming from react, htmx introduces a completely different application paradigm. It requires going
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:277]:
```suggestion
Coming from React, htmx introduces a completely different application paradigm. It requires going
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+back to the basics, which felt good. Simply not having the app written in React felt great.
+
+Also the combination of [Astro DB](https://astro.build/db/) with [Astro
+SSR](https://docs.astro.build/en/guides/server-side-rendering/) hosted on
+[Netlify](https://www.netlify.com/) was great. After it initially seemed like an anti-pattern to
+use Astro for dynamic content, the site itself is quite fast and the experience was great.
+
+Htmx is conceptionally easy to grasp, and while the initial learning curve is somewhat steep, it
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:285]:
```suggestion
Htmx is conceptually easy to grasp, and while the initial learning curve is somewhat steep, it
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+JSON api middleman can be skipped, by shipping HTML directly. Especially with Astro, having
+`.astro` templates all the way through, with minimal external dependencies, keeps the complexity
+low.
+
+On the other hand, building this app highlighted also a few strengths of React. Handling the data
+flow between components is the main purpose of the React component properties. With htmx, doing
+the same while only fetching data once, had us come up with some last-mile JavaScript. The
+alternative could be to simply fetch more HTML, which seems to be part of the htmx paradigm.
+
+Generally the chance of writing explicit JavaScript seems slightly higher with htmx, but at-least
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:271]:
```suggestion
Generally the chance of writing explicit JavaScript seems slightly higher with htmx, but at least
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+> This is the year of HTMX!
+
+
+While the htmx examples in the [documentation](https://htmx.org/examples/) look easy enough, we
+decided to take it to the test and re-implemented an existing [React
+project](https://github.com/flyck/party-task-planner) in
+[htmx](https://github.com/flyck/astro-party).
+
+We had created the original app earlier this year, mostly to play around with Websockets and
+AppSync subscriptions. In the frontend it was based on React and Next.js, hosted on top of
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:22]:
```
In the frontend, it was based on React and Next.js, hosted on Vercel. The backend ran on AWS AppSync and DynamoDB for persistent storage. In the re-implementation, we moved the frontend to pure HTML and JavaScript using htmx and Astro. For the backend, we used Astro DB and Astro SSR, hosted on Netlify.
```
Rewrite suggestion.

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 11]
```diff
@@ -0,0 +1,289 @@
+---
+publishDate: "May 23 2024"
+title: "Ditch React: Build Faster with htmx, Astro DB & Astro SSR in 2024"
+description: "Explore common techniques and solutions needed for building a basic CRUD app with this modern react-less stack."
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:4]:
```suggestion
description: "Explore common techniques and solutions for building a fast and efficient CRUD app using htmx, Astro DB, and Astro SSR without React in 2024."
```

File: data/blog/howto-htmx-astrodb-astrossr.md [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+[re-implementation](https://github.com/flyck/astro-party), we moved the frontend to pure HTML and
+js, using [htmx](https://htmx.org/ "Visit HTMX Official Site for more details") and
+[Astro](https://astro.build). For the backend we used [Astro DB](https://astro.build/db/), which
+is currently in early preview, and Astro SSR hosted on [Netlify](https://www.netlify.com/).
+
+We were curious how well this stack could could work for CRUD apps, but especially the experience
+of using htmx instead of React. In the following we'll share basic techniques for this stack,
+and also our conclusions, looking back.
+
+## Responsive Buttons
```
Comment (hendriknielaender) [inline data/blog/howto-htmx-astrodb-astrossr.md:33]:
```suggestion
## Table of Contents
- [Creating Responsive Buttons with htmx and Tailwind CSS](#creating-responsive-buttons-with-htmx-and-tailwind-css)
- [Implementing Skeletons for Better UX](#implementing-skeletons-for-better-ux)
- [Inline Validation for Immediate Feedback](#inline-validation-for-immediate-feedback)
- [Handling Errors with Toast Notifications](#handling-errors-with-toast-notifications)
- [Enhancing Navigation with View Transitions](#enhancing-navigation-with-view-transitions)
- [Using Astro SSR & Netlify for Hosting](#using-astro-ssr--netlify-for-hosting)
- [Is htmx the Ultimate Choice?](#is-htmx-the-ultimate-choice)
- [Conclusion](#conclusion)

## Creating Responsive Buttons with htmx and Tailwind CSS
```

**Comment by flyck** [comment]
Comic thumbnail it is 🙌🏻 then I guess we're good to go? 😃

**Comment by hendriknielaender** [comment]
> Comic thumbnail it is 🙌🏻 then I guess we're good to go? 😃 

Yes ✅ feel free to merge

**Pull request merged**